Bottom: 1c8fbc5fdc8ac86358d7095d9ac4561f47eec4ea
Top:    6824bd04c82761ee7e8d74496b6527b2166fd460
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-16 14:45:34 -0700

Refresh of fork

---

diff --git a/src/preload.c b/src/preload.c
index 177c621..9c4c423 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -431,12 +431,10 @@ static int connect_fork(int socket, const struct sockaddr *addr, socklen_t addrl
 	if (ret != sizeof msg)
 		return ret;
 
-	rs = rsocket(domain, type, protocol);
-
-	real_shutdown(fd, SHUT_RDWR);
-	real_close(fd);
-
-	fd_store(socket, fd, fd_rsocket);
+	ret = transpose_socket(socket, &fd, fd_rsocket, rsocket,
+			       real_close, rclose, real_getsockname,
+			       real_getsockopt, rsetsockopt,
+			       real_fcntl, rfcntl);
 	return connect(socket, addr, addrlen);
 }
